### PR TITLE
Cosmo io private

### PR DESCRIPTION
**Pull request by nstarman**
It says in multiple ways that ``cosmology.io`` is private: in its absence in the docs, in ``cosmology.io.__init__.py`` and that it's not in ``cosmology.__init__::__all__``. This PR helps enforce that.

Also, this PR does some small improvements: renaming arguments and cleaning up imports.

### CODE REVIEW

File: astropy/cosmology/_io/mapping.py [hunk 1]
```diff
@@ -17,12 +17,12 @@
 __all__ = []  # nothing is publicly scoped
 
 
-def from_mapping(map, *, move_to_meta=False, cosmology=None):
+def from_mapping(mapping, /, *, move_to_meta=False, cosmology=None):
```
Comment (eerovaher) [inline astropy/cosmology/_io/mapping.py:20]: What does this have to do with renaming `astropy.cosmology.io`?

File: astropy/cosmology/tests/test_utils.py [hunk 2]
```diff
@@ -3,10 +3,7 @@
 import numpy as np
 import pytest
 
-from astropy.cosmology.utils import (
-    aszarr,
-    vectorize_redshift_method,
-)
+from astropy.cosmology.utils import aszarr, vectorize_redshift_method
```
Comment (eerovaher) [inline astropy/cosmology/tests/test_utils.py:6]: I prefer the updated code, but this change is unrelated to what the pull request claims to be doing.

**Review by eerovaher — CHANGES_REQUESTED** [review]
If `astropy.cosmology.io` is private then changing its name to make it clearer is a good idea. However, this pull request seems to contain unrelated changes and the documentation build failure looks to be related.

File: astropy/cosmology/_io/mapping.py (hunk 1 above)
Comment (nstarman) [reply astropy/cosmology/_io/mapping.py:20]: General improvement. These are private functions and can be changed. In this case it's changing from overriding a builtin to a better name.

File: astropy/cosmology/tests/test_utils.py (hunk 2 above)
Comment (nstarman) [reply astropy/cosmology/tests/test_utils.py:6]: I deleted 1 comma and let black do it's thing.

File: astropy/cosmology/_io/mapping.py (hunk 1 above)
Comment (eerovaher) [reply astropy/cosmology/_io/mapping.py:20]: This change to the code is an improvement, I just didn't expect it because the pull request description only talks about renaming `astropy.cosmology.io`. Could you edit the pull request so that it would describe all the changes?

File: astropy/cosmology/tests/test_utils.py (hunk 2 above)
Comment (nstarman) [reply astropy/cosmology/tests/test_utils.py:6]: I wonder if ruff has a tool to tell if end commas are superfluous...

File: astropy/cosmology/_io/mapping.py (hunk 1 above)
Comment (nstarman) [reply astropy/cosmology/_io/mapping.py:20]: Done!

File: astropy/cosmology/tests/test_utils.py (hunk 2 above)
Comment (eerovaher) [reply astropy/cosmology/tests/test_utils.py:6]: Yes, rule [COM819](https://beta.ruff.rs/docs/rules/prohibited-trailing-comma/). But it would interfere with [Black's magic trailing comma](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#the-magic-trailing-comma) so I am in not in favor of enforcing that rule.

File: astropy/cosmology/_io/tests/test_ecsv.py [hunk 3]
```diff
@@ -3,9 +3,10 @@
 # THIRD PARTY
 import pytest
 
+from astropy.cosmology._io.ecsv import read_ecsv, write_ecsv
+
 # LOCAL
```
Comment (eerovaher) [inline astropy/cosmology/_io/tests/test_ecsv.py:8]: The comments here are lying - renaming a local module did not turn it into a third party module.

File: astropy/cosmology/_io/tests/test_latex.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 import pytest
 
 # LOCAL
-from astropy.cosmology.io.latex import _FORMAT_TABLE, write_latex
+from astropy.cosmology._io.latex import _FORMAT_TABLE, write_latex
```
Comment (eerovaher) [inline astropy/cosmology/_io/tests/test_latex.py:7]: At least here the comment is not lying, but I still don't think it's needed.

**Review by eerovaher — CHANGES_REQUESTED** [review]
There are quite a few instances where comments label `astropy.cosmology._io` as a third party module, which is not true. I have highlighted one of them, but there are more.  I don't think the `# THIRD PARTY` and `# LOCAL` comments provide much benefit anyways, at least not if we are using Ruff to keep the imports sorted. It would be simplest to just remove these comments.

Furthermore, f70ba0c3a8a33f54d92fcf812cd1130da0cddb8a should be merged with the commit that broke the documentation build.

File: astropy/cosmology/_io/tests/test_ecsv.py (hunk 3 above)
Comment (nstarman) [reply astropy/cosmology/_io/tests/test_ecsv.py]: Agreed. Removing all instances of the import location comments in ``cosmology/_io``.

File: astropy/cosmology/realizations.py [hunk 5]
```diff
@@ -10,6 +10,7 @@
 from astropy.utils.decorators import deprecated
 from astropy.utils.state import ScienceState
 
+from . import _io  # noqa: F401
```
Comment (eerovaher) [inline astropy/cosmology/realizations.py:13]: Ruff rule [F401](https://beta.ruff.rs/docs/rules/unused-import/) is about unused imports. It's worth adding a comment here why `_io` is being imported if it is unused. The reason is not obvious from the code.

**Review by eerovaher — CHANGES_REQUESTED** [review]
This is mostly good, but I do have a question about an unused import.

File: astropy/cosmology/realizations.py (hunk 5 above)
Comment (nstarman) [reply astropy/cosmology/realizations.py]: Done!

File: astropy/cosmology/realizations.py [hunk 6]
```diff
@@ -10,6 +10,7 @@
 from astropy.utils.decorators import deprecated
 from astropy.utils.state import ScienceState
 
+from . import _io  # Register I/O methods for reading builtin realizations  # noqa: F401
```
[the remark this answers is not in the document]
Comment (nstarman) [reply astropy/cosmology/realizations.py:13]: I would like to replace the I/O registration with entry-points, but that will require a PR to `io/registry`.

[the remark this answers is not in the document]
Comment (eerovaher) [reply astropy/cosmology/realizations.py:13]: > When trying to read the in-built realizations, it doesn't know about any I/O methods unless `_io` has been imported first.

So is it the case that the I/O method registration requires a specific module import order and the line here is meant to enforce that? If so then that's what the comment should say.

[the remark this answers is not in the document]
Comment (nstarman) [reply astropy/cosmology/realizations.py]: The import order importance is that `._io` is registered before `.realizations`.

**Review by eerovaher — APPROVED** [review]
Looks good to me, but I'll leave the pull request open for a while in case @WilliamJamieson or @mhvk want to have a look at it.

**Comment by mhvk** [comment]
I still don't really like having private modules (unless they are bindings to C code or so) hence felt I should not review this, as really what counts is not my opinion but whether it works better for `cosmology`.

**Comment by eerovaher** [comment]
I think anybody that would have wanted to say something has had plenty of time to do so, there's no need to delay this any further.

**Comment by nstarman** [comment]
Thanks @eerovaher for the multiple detailed reviews!
